src/libraries/Common/src/Interop/Unix/Interop.IOErrors.cs

```diff
@@ -114,6 +114,7 @@ internal static Exception GetExceptionForIoErrno(ErrorInfo errorInfo, string? pa
         switch (errorInfo.Error)
         {
             case Error.ENOENT:
+            case Error.ENOTDIR:
```

Comment:
Changing this cause a different type exception and different exception message be used.

`ENOTDIR` and `ENOENT` are distinct errors.
`ENOTDIR` means something expected is not a directory.
We shouldn't treat them the same unless there is a reason to do so (like Windows compatibility).

What caused you to update this? Was it a test failure?

Reply:
I wasn't sure if it should go there or not, the reason I added it is because it is on the list of errors that `setattrlist` can throw. What should I throw instead?

Reply:
You don't need to add it, it will use the `default` case.

```cs
            default:
                return GetIOException(errorInfo);
```

Reply:
Okay, I'll revert that in the next commit.